### PR TITLE
Display maximum download size before install

### LINKED ISSUE #47: Show total estimated size before installing
For reference, this is what the CLI does:

```
       ID                                    Arch    Branch  Remote   Download
 1.     org.gnome.Platform                    x86_64  3.28    flathub  < 481.8 MB
 2.     org.freedesktop.Platform.VAAPI.Intel  x86_64  1.6     flathub    < 2.7 MB
 3.     org.freedesktop.Platform.ffmpeg       x86_64  1.6     flathub    < 2.9 MB
 4.     org.gnome.Platform.Locale             x86_64  3.28    flathub  < 188.9 MB (partial)
 5.     org.gnome.Epiphany                    x86_64  stable  flathub    < 3.4 MB
 6.     org.gnome.Epiphany.Locale             x86_64  stable  flathub    < 2.0 MB (partial)

Proceed with these changes to the user installation? [Y/n]: 
```

That's probably way too much data and we can just do something like we do in AppCenter with the “up to” and help text on tooltip.

### PR DESCRIPTION
**Pull request by davidmhewitt**
Fixes #47 

Needs a bit of design work to decide how we want to show this information. Feel free to iterate on this branch or give me suggestions as you see fit. Worth bearing in mind we'll want to stick a list of sandbox permissions in here too somewhere/somehow.

I also took the liberty to rework my last commit slightly as I needed to perform a dry run of the install to get the size of all the non-installed deps too. Makes it a little harder to review, but this is still in a good place for adding the display of required sandbox permissions too. We also gain an added benefit of not having to wait for the user to click the install button to tell them that it's already installed.

### CODE REVIEW

**Comment by davidmhewitt** [comment]
@danrabbit All good points, thanks! I've made some progress towards that but may need some design work in places.

**Comment by danrabbit** [comment]
Screenshots for discussion purposes:

Screenshot from 2019-10-08 07 57 19@2x
Screenshot from 2019-10-08 07 56 54@2x
Screenshot from 2019-10-08 07 57 07@2x

**Comment by cassidyjames** [comment]
Personally I think moving to the success page for the "already installed" case is fine because:

- The user wanted to install the app. It's installed.
- The user might still want to open the app, and we show that button on success.
- It's not an error, it's just a no-op.

**Comment by danrabbit** [comment]
@cassidyjames yeah you're right. Reverted that change :)

I feel pretty good about this now. @cassidyjames any more thoughts?

File: src/MainWindow.vala [hunk 1]
```diff
@@ -62,6 +62,17 @@ public class Sideload.MainWindow : Gtk.ApplicationWindow {
...
         file.installation_failed.connect (on_install_failed);
         file.installation_succeeded.connect (on_install_succeeded);
+        file.details_ready.connect (() => {
+            if (file.already_installed) {
+                var success_view = new SuccessView (SuccessView.SuccessType.ALREADY_INSTALLED);
+
+                stack.add (success_view);
+                stack.visible_child = success_view;
+            } else {
+                main_view.display_details (file.download_size, file.already_installed);
```
Comment (davidmhewitt) [inline src/MainWindow.vala]: Don't need to pass `already_installed` to this method anymore :slightly_smiling_face:

**Pull request merged**